pkg/scheduler/scheduler.go

```diff
@@ -106,7 +106,7 @@ func (s *Scheduler) schedule(ctx context.Context) {
 		log := log.WithValues("queuedWorkload", klog.KObj(e.Obj), "clusterQueue", klog.KRef("", e.ClusterQueue))
 		if err := s.admit(ctrl.LoggerInto(ctx, log), &e); err != nil {
 			log.Error(err, "Failed admitting workload by clusterQueue")
-			err := workload.UpdateWorkloadStatus(ctx, s.client, e.Obj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse, "Pending", err.Error())
+			err := workload.UpdateWorkloadStatus(ctx, s.client, e.Obj, s.recorder, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse, "Pending", err.Error())
```

Comment:
in kube-scheduler, we only create an event if the workload didn't fit, not when there is an error. Errors are usually conflicts on the resource version, or we are trying to schedule something that we already admitted. These errors are internal, in the sense that the user doesn't need to do anything to change the situation (the system will recover).

On the other hand, if the workload doesn't fit, the user could do something, like terminate some running jobs, or request more quota to the admin, etc.
That would be lines 124-128.

Reply:
Got it, did a re do of the PR, now looks...much smaller